### PR TITLE
Wip/6.0 reenable tests 5

### PR DESCRIPTION
**Pull request by dreab8**

### CODE REVIEW

**Comment by sebersole** [comment]
Do we still need this?  Sorry it took me so long to look

**Comment by dreab8** [comment]
the PR makes a lazy collection with  `fetch=join` (<set name="bids" cascade="all" fetch="join" ....>) loaded lazily when a Criteria or HQL query does not explicitly fetch it

**Comment by beikov** [comment]
Could you rebase this PR @dreab8? I pushed my fixes to main already for retaining laziness even if fetch="join" is used.

**Comment by dreab8** [comment]
@beikov rebased :)

File: hibernate-core/src/main/java/org/hibernate/query/sqm/sql/BaseSqmToSqlAstConverter.java [hunk 1]
```diff
@@ -6215,6 +6216,12 @@ else if ( getLoadQueryInfluencers().hasEnabledFetchProfiles() ) {
 						}
 					}
 				}
+				else if ( fetchable instanceof PluralAttributeMapping && fetchable.getMappedFetchOptions()
+						.getStyle() == FetchStyle.JOIN ) {
+					if ( ( (PluralAttributeMapping) fetchable ).isLazy() ) {
+						fetchTiming = FetchTiming.DELAYED;
+					}
+				}
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/query/sqm/sql/BaseSqmToSqlAstConverter.java:6224]: I don't think this is necessary anymore

File: hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/PluralAttributeMappingImpl.java [hunk 2]
```diff
@@ -100,6 +100,8 @@
 
 	private final String bidirectionalAttributeName;
 
+	private final boolean isLazy;
+	private final CascadeStyle cascadeStyle;
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/PluralAttributeMappingImpl.java:104]: I don't think this is necessary anymore

File: hibernate-core/src/main/java/org/hibernate/metamodel/mapping/PluralAttributeMapping.java [hunk 3]
```diff
@@ -116,4 +116,6 @@ default void applyWhereRestrictions(
 			SqlAstCreationState creationState) {
 		getCollectionDescriptor().applyWhereRestrictions( predicateConsumer, tableGroup, useQualifier, creationState );
 	}
+
+	boolean isLazy();
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/metamodel/mapping/PluralAttributeMapping.java:120]: I don't think this is necessary anymore

**Review by beikov — COMMENTED** [review]
Looks good mostly, but I think a few changes have now become unnecessary

**Review by beikov — APPROVED** [review]

**Pull request merged**